File: 2022/phase2_dress_rehearsal/byuvrx/src/frostlab-inekf-54148f8ac0c7/python/tests/Inertial/test_DVLSensor.py
import numpy as np
from numpy.testing import assert_allclose
from inekf import SE3, DVLSensor
import pytest

def test_processZ():
    state = SE3[2,6]()
    # append IMU values to the end of it
    z = np.array([1,2,3,0,0,0])

    ds = DVLSensor()
    z_full = ds.processZ(z, state)

    expected = np.array([1,2,3,-1,0])
    assert_allclose(expected, z_full)

File: 2022/phase2_dress_rehearsal/byuvrx/src/frostlab-inekf-54148f8ac0c7/tests/Inertial/test_DVLSensor.cpp
#include "gtest/gtest.h"
#include <Eigen/Core>
#include <InEKF/Core>
#include <InEKF/Inertial>

#define EXPECT_MATRICES_EQ(M_actual, M_expected) \
  EXPECT_TRUE(M_actual.isApprox(M_expected, 1e-6)) << "  Actual:\n" << M_actual << "\nExpected:\n" << M_expected


TEST(DVLSensor, processZ){
    InEKF::SE3<2,6> state;
    Eigen::Matrix<double,6,1> z;
    z << 1,2,3,0,0,0;

    InEKF::DVLSensor ds;
    Eigen::Matrix<double,5,1> z_full = ds.processZ(z, state);

    Eigen::Matrix<double,5,1> expected;
    expected << 1, 2, 3, -1, 0;

    EXPECT_MATRICES_EQ(z_full, expected);
}